templates/Makefile.erb

```diff
@@ -79,8 +79,11 @@ cleanup-components: <%= @components.map {|comp| "#{comp.name}-cleanup" }.join("
 	touch <%= comp.name %>-patch
 
 <%= comp.name %>-configure: <%= comp.name %>-patch <%= list_component_dependencies(comp).join(" ") %>
+	<%- if comp.get_build_dir -%>
+	[ -d <%= comp.get_build_dir %> ] || mkdir <%= comp.get_build_dir %>
```

Comment:
Ugh, I bet I was using some variant `mkdir` (bsd, busybox, whatever) and it just exited successfully. Disregard nitpick; safety first!